### DISCUSSION TITLE
Grand piano

### DISCUSSION THREAD

**Discussion by matiasmorant** [marcel-licence/esp32_basic_synth#95] [General]
Can it produce the sound of an acoustic piano?
Maybe by sampling or some complex FM synth configuration?

**Comment by marcel-licence** [comment]
Maybe the sampler project or the project emulating a mellotron. There are some limitations because piano samples require a lot of memory. I was able to emulate the super mario piano (SNES)

**Comment by roguestreak** [comment]
How much space is available Marcel?  The venerable Ensoniq Mirage sampler used to do quite good acoustic piano samples and that had a whopping 2x 64kB of memory (64kB per keyboard "half", 128kB total), IIRC it used perhaps 4 distinct samples per "half", so let's say 16kB per sample.  They were 8bit and very low sample rate (guessing maybe 16kb/s?), but they were extremely optimised to get the most out of it.
I just went searching and found a whole bunch of the original wav files in zips in this post, haven't been through them yet but may be a good starting point?
https://gearspace.com/board/electronic-music-instruments-and-electronic-music-production/1297781-every-ensoniq-mirage-sample-ancient-ftp.html

An example of a piano, not sure what disk it's on;
https://www.youtube.com/watch?v=CJR0eMg16zQ&t=110s

**Reply by marcel-licence** [reply]
128k should work. The ESP32 has about 500k but not all of the memory can be used because it is needed for internal stuff and also the sampler. 
The wav files would help. I could imagine that an integration is possible. Unfortunately some links seem to be dead. If you find a direct link to the samples please let me know.

**Reply by roguestreak** [reply]
Post #18 has zip files attached which work, I've just successfully downloaded them and there's a heap of disks worth and most aren't named, I'll need to find a faster method of listening to all of the subfolders to try to track anything down.
https://gearspace.com/board/showpost.php?p=14518906&postcount=18
I cut my teeth on the Mirage, but it's been a looong time since I booted mine up.  Both my wroom32 and rp2040 arrived the other day so hopefully over the coming weeks I can attempt to get them operational, I'll start with the synth but if I can work out the sampler all the better, particularly if it can be used as a sampling synthesizer, with for example start sound samples which then merge to synthesis for the later stages of the sound.
I'll see if I can track down which particular sample(s) the piano is, if it is indeed in that collection.

**Reply by roguestreak** [reply]
Looks like "Disk01" in the first collection has the pianos (I think there are three or more types), with "Kboards1" in the second collection being the same set.
I found if I "add folder" in VLC it will bring in the subfolders and play through them all.  I didn't go through all of the FX folders etc., there's a hell of a collection in there and it's all very tiny; we used to have to save three sets (six banks) of samples onto a single-sided single-density floppy, ie. 360kB of storage.

**Reply by marcel-licence** [reply]
I can agree the required storage is quite low. One confusing thing is what the names are meaning. Some samples are very short. In addition to that I do not know which part should be looped. When running the sampler without effects etc. it should be possible to load them into the normal RAM for playback.

**Comment by matiasmorant** [comment]
I saw there are 3 realistic piano presets in Sytrus plugin of FL Studio, they are made only with FM synthesis.

**Reply by marcel-licence** [reply]
@matiasmorant fortunately I have FL Studio and was able to try out these sounds. It looks like there is some intensive additive synthesis involved for these sounds. The rhodes sound sounds interesting to me. In addition to that I do not really know yet how the Sytrus plugin works. 
I was able to create a sample based piano on the DaisySeed some time ago: https://www.youtube.com/watch?v=nOVG7N4dujE but this requires a lot of memory

**Reply by matiasmorant** [reply]
we are talking about the sytrus pianos by "daven hughes", right? (his name is in the "main" tab)
Sytrus is a generic FM/AM synth. I can teach you how it works if you want, maybe you can get some ideas to implement on esp32.
That video sounds great.
